docker/Dockerfile

```diff
@@ -15,4 +15,15 @@ RUN apk add --no-cache ca-certificates fuse openssh-client tzdata jq
 
 COPY --from=builder /go/src/github.com/restic/restic/restic /usr/bin
 
-ENTRYPOINT ["/usr/bin/restic"]
+# IO class default is "none"- 0, however busybox ionice `-c0 -n<something>`
+# since priority has no meaning for no scheduler.
+# Thus the set logic below is necessary.
+ENV IONICE_CLASS=
+ENV IONICE_PRIORITY=4
+ENV NICE=0
+
+ENTRYPOINT ["/bin/sh", "-ec", "\
+    set -- nice -n \"${NICE}\" /usr/bin/restic \"$@\";  \
+    [ -n \"${IONICE_CLASS}\" ] && set -- ionice -c \"${IONICE_CLASS}\" -n \"${IONICE_PRIORITY}\" \"$@\"; \
+    exec \"$@\"; \
+    "]
```

Comment:
This still loses the command arguments:
```
docker/build.sh
docker run --rm docker.io/restic/restic:latest version
```

Prints the help output and not the version.

```
docker run --rm docker.io/restic/restic:latest version version
```

Prints the version, but that's obviously not how this should look like. So, you'll have to add `$0` in addition to `$@`.

At this point I'd prefer to just add an entrypoint script that handles everything necessary, the current inline calls to `sh` is already nearly unreadable.